src/libpgmoneta/management.c

```diff
@@ -3409,3 +3445,301 @@ print_info_json(struct json* json)
+		{
+			goto error;
+		}
+	
+		pgmoneta_json_item_put_string(output_item, "Actual_Checksum", actual_checksum);
+	
+		if (read_string("pgmoneta_management_read_verify_data", socket, &path))
+		{
+			goto error;
+		}
+	
+		pgmoneta_json_item_put_string(output_item, "Actual_Checksum", path);
+	
+		//put to array
+		pgmoneta_json_array_append_object(output_array, output_item);
+	
+		//end
+		pgmoneta_json_item_put_object(verify_output, "verify_output", output_array);
+
+		read_int32("pgmoneta_management_read_verify_data", socket, &should_read);
+	}
+
+	// free(path);
+	// free(manifest_checksum);
+	// free(actual_checksum);
+	// free(check);
+
+	return json;
+
+
+error:
+
+	free(path);
+	free(manifest_checksum);
+	free(actual_checksum);
+	free(check);
+
+	if (json != NULL)
+	{
+		pgmoneta_json_set_command_object_faulty(json, strerror(errno));
+	}
+	
+	errno = 0;
+
+	return json;
+}
+
+int
+pgmoneta_management_write_verify_data(int socket, int srv, char* backup_id, char* option) 
+{
+	char* d = NULL;
+	char* backup_manifest = NULL;
+	char* backup_label = NULL;
+	int backup_index = -1;
+	int number_of_backups = 0;
+	bool report_all;
+	struct configuration* config;
+	struct backup** backups = NULL;
+
+	if (option == NULL || strcmp(option, "fail") == 0)
+	{
+		report_all = false;
+		pgmoneta_log_debug("Verify: report only failed checksums check");
+	}
+	else
+	{
+		report_all = true;
+		pgmoneta_log_debug("Verify: report all checksums check");
+	}
+
+	config = (struct configuration*)shmem;
+
+	//backup dir for server;
+	d = pgmoneta_get_server_backup(srv);
+
+	//get all backups for the server;
+	if (pgmoneta_get_backups(d, &number_of_backups, &backups) != 0)
+	{
+		goto error;
+	}
+	free(d);
+	d = NULL;
+
+	//select proper backup
+	if (!strcmp(backup_id, "oldest"))
+	{
+		for (int i = 0; backup_index == -1 && i < number_of_backups; i++)
+		{
+			if (backups[i] != NULL)
+			{
+				backup_index = i;
+			}
+		}
+	}
+	else if (!strcmp(backup_id, "latest") || !strcmp(backup_id, "newest"))
+	{
+		for (int i = number_of_backups - 1; backup_index == -1 && i >= 0; i--)
+		{
+			if (backups[i] != NULL)
+			{
+				backup_index = i;
+			}
+		}
+	}
+	else
+	{
+		for (int i = 0; backup_index == -1 && i < number_of_backups; i++)
+		{
+			if (backups[i] != NULL && !strcmp(backups[i]->label, backup_id))
+			{
+				backup_index = i;
+			}
+		}
+	}
+
+	//backup not found
+	if (backup_index == -1)
+	{
+		pgmoneta_log_error("Verify: No identifier for %s/%s", config->servers[srv].name, backup_id);
+		goto error;
+	}
+
+	//backup not valid
+	if (backups[backup_index]->valid == VALID_FALSE)
+	{
+		pgmoneta_log_error("Verify: Backup is not valid");
+		goto error;
+	}
+
+
+	//check if backup_manifest exists and is a regular file
+	backup_manifest = pgmoneta_get_server_backup_identifier_data(srv, backups[backup_index]->label);
+	backup_manifest = pgmoneta_append(backup_manifest, "backup_manifest");
```

Comment:
Note that we now have our own manifest, which locates in the same dir where data/ and tablespaces are. The original manifest is now subject to encryption and compression so you cannot read it directly. Our version (the name is backup.manifest) is in csv format, which we also have support for, so you can read it relatively more easily. Check out [wf_manifest](https://github.com/pgmoneta/pgmoneta/blob/d9eaf72529d6c40c992208a9c61b9eb2a055b3e9/src/libpgmoneta/wf_manifest.c#L69) and [pgmoneta_compare_manifests](https://github.com/pgmoneta/pgmoneta/blob/d9eaf72529d6c40c992208a9c61b9eb2a055b3e9/src/libpgmoneta/manifest.c#L128) for details.

Reply:
I wondered how I should deal with encrypted and compressed manifests and files, but assumed, for simplicity sake, that users would have to run `pgmoneta-cli decrypt` and decompress a backup before running `pgmoneta-cli`.

After this comment I will look on the `backup.manifest`. Also, I will need to refactor to account for its format not being json.

Reply:
@jesperpedersen Can you clarify if I should work with the json `backup_manifest` or the pgmoneta native `backup.manifest` csv file? And should the verify feature also decrypt and decompress manifests when needed - or I should assume the user will do these operations before running `pgmoneta-cli verify <server> <id>`?

Reply:
> I wondered how I should deal with encrypted and compressed manifests and files,

Yeah, that's a good point. I wonder if we should decompress and decrypt for user, or at least make it an option for user when verify. 

> After this comment I will look on the backup.manifest. Also, I will need to refactor to account for its format not being json.

No worries. This won't be too much changes, csv is much easier format than json.